### ISSUE TITLE
Release 1.1.0 issue

### ISSUE THREAD

**Issue by code2prog** [antonioribeiro/google2fa-laravel#69]
Something is not right with latest release on laravel 6. After upgrade package to version 1.1.0 i get error like this: 
`Call to undefined method Illuminate\Support\MessageBag::getBag()`

it looks like issue with @error blade tag and session messages. After back to 1.0.1 everything works fine.

**Comment by willbrowningme** [comment]
Similar issue here. After the upgrade the 2FA view is being returned with errors even when it is first displayed (before the user has entered an incorrect code).

It appears to be due to this pull request #63 that removed the check for the OTP input.

**Comment by AdamKyle** [comment]
Can this please be fixed, I am unable to upgrade to laravel 6 because my tests fail due the issues stated here.

**Comment by AdamKyle** [comment]
Update, best possible solution is to downgrade to 1.0.1, it works with laravel 6.

**Comment by antonioribeiro** [comment]
Trying to understand the source of the problem. How do I reproduce it?

@willbrowningme, that PR was merged because the CheckOTP method was already calling inputHasOneTimePassword():

![image](https://user-images.githubusercontent.com/3182864/64891068-521a7b00-d647-11e9-9cb5-cbb9ae44c75a.png)

**Comment by willbrowningme** [comment]
@antonioribeiro yes I saw that was the reason for the PR, however removing that check changes the response returned by `makeStatusCode()`.

Without the `$this->inputHasOneTimePassword()` check in `makeStatusCode()` then 

```php
return !$this->checkOTP()
            ? SymfonyResponse::HTTP_UNPROCESSABLE_ENTITY
            : SymfonyResponse::HTTP_OK;
```

Evaluates to true and returns `SymfonyResponse::HTTP_UNPROCESSABLE_ENTITY` instead of `SymfonyResponse::HTTP_OK`. So when the input does not have a OTP it returns 422 with errors.

**Comment by antonioribeiro** [comment]
@willbrowningme, looking at the original code, I feel it was wrong since the beginning:

```
protected function makeStatusCode()    
{       
    $this->inputHasOneTimePassword() && !$this->checkOTP()    
        ? SymfonyResponse::HTTP_UNPROCESSABLE_ENTITY          
        : SymfonyResponse::HTTP_OK; 
}
```

IF the input is filled AND the OTP is wrong it will return HTTP_OK, which is not correct. With this behaviour all OTPs are returning as correct.

This should be the correct:

```
protected function makeStatusCode()    
{       
    !$this->inputHasOneTimePassword() || !$this->checkOTP()    
        ? SymfonyResponse::HTTP_UNPROCESSABLE_ENTITY          
        : SymfonyResponse::HTTP_OK; 
}
```

IF the input is filled OR the OTP is wrong it must return HTTP_UNPROCESSABLE_ENTITY.

**Comment by willbrowningme** [comment]
@antonioribeiro but then what about when the user has just logged in and they are shown the 2FA view where they must enter their OTP for the first time?

The input will not have the OTP as the user has not had chance to enter it yet so the view will be shown with errors and a 422 status code.

**Comment by asbiin** [comment]
I can confirm the bug.

Before #63, the first hit after a login (`Response::makeRequestOneTimePasswordResponse`) returned a page where the user is able to enter the validation code.
Now as `makeStatusCode()` returns HTTP_UNPROCESSABLE_ENTITY, because `inputHasOneTimePassword ` is false, the user sees the error "The two factor authentication has failed." ... but as it's the first time the page is displayed, that's not a normal behavior.

**Comment by asbiin** [comment]
To be clear, the right answer with the `makeStatusCode()` function should be:

 input has OTP:|YES|NO
----------------|----|---
OTP is OK|true|true
OTP is Wrong|false|true

`makeStatusCode()` is only used in `makeRequestOneTimePasswordResponse()` so I don't see a problem with the previous code.

-> if the user did'nt enter an OTP, no error should be displayed with `makeRequestOneTimePasswordResponse()`, so `makeStatusCode()` return HTTP_OK.
-> if the user enter an OTP, response of `makeStatusCode()` depends on the validation of the OTP

@antonioribeiro you said about the previous code of `makeStatusCode()`:
```php
protected function makeStatusCode()    
{       
    return $this->inputHasOneTimePassword() && !$this->checkOTP()    
        ? SymfonyResponse::HTTP_UNPROCESSABLE_ENTITY          
        : SymfonyResponse::HTTP_OK; 
}
```

> IF the input is filled AND the OTP is wrong it will return HTTP_OK, which is not correct. With this behaviour all OTPs are returning as correct.

That is not correct.
If the input is filled, the OTP is checked right with `verifyOneTimePassword()` function, as it was checked before, and it returns HTTP_UNPROCESSABLE_ENTITY  if if fails, as expected.

My advice is to revert #63.

**Comment by asbiin** [comment]
This one is fixed by #82
